BaragonUI/app/components/services/Services.jsx

```diff
@@ -1,21 +1,182 @@
+    className="keep-in-check"
+    cellData={
+      (rowData) => rowData.service.serviceBasePath
+    }
+    sortable={true}
+  />
+);
+
+const GroupsColumn = (
+  <Column
+    label="Groups"
+    id="groups"
+    key="groups"
+    cellData={
+      (rowData) => {
+        return rowData.service.loadBalancerGroups.map((groupId) => {
+          return (
+            <Link
+              to={`group/${groupId}`}
+              title={`Details for ${groupId}`}
+              key={groupId}
+            >
+            {groupId}
+            </Link>
+          );
+        });
+      }
+    }
+    sortable={false}
+  />
+);
+
+const UpstreamsColumn = (
+  <Column
+    label="Upstreams"
+    id="upstreams"
+    key="upstreams"
+    cellData={
+      (rowData) => rowData.upstreams.length
+    }
+    sortable={true}
+  />
+);
+
+const ButtonsColumn = (
+  <Column
+    label=""
+    id="actions"
+    key="actions"
+    className="actions-column"
+    cellRender={
+      (cellData, rowData) => {
+        const deleteService = (
+          <DeleteServiceButton serviceId={rowData.service.serviceId} />
+        );
+
+        const reloadService = (
+          <ReloadServiceButton serviceId={rowData.service.serviceId} />
+        );
+
+        const removeTooltip = (
+          <Tooltip id="removeUpstreams">
+            Remove all upstreams from this service
+          </Tooltip>
+        );
+
+        const removeUpstreams = (
+          <RemoveUpstreamsButton loadBalancerService={cellData.service} upstreams={cellData.upstreams}>
+            <OverlayTrigger placement="top" overlay={removeTooltip}>
+              <a>
+                <span className="glyphicon glyphicon-remove-circle" />
+              </a>
+            </OverlayTrigger>
+          </RemoveUpstreamsButton>
+        );
+
+        const onlyIfUpstreams = (button) => {
+          if (cellData.upstreams.length > 0) {
+            return button;
+          }
+
+          return null;
+        }
+
+        return (
+          <div className="hidden-xs">
+            {onlyIfUpstreams(reloadService)}
+            {onlyIfUpstreams(removeUpstreams)}
+            {deleteService}
+            <JSONButton className="inline" object={cellData} showOverlay={true}>
+              {'{ }'}
+            </JSONButton>
+          </div>
+        );
+      }
+    }
+  />
+);
 
 class Services extends Component {
+  static propTypes = {
+    services: PropTypes.array,
+    allowEdit: PropTypes.bool,
+  };
+
   render() {
+    if (this.props.allowEdit) {
+      return (
+        <div>
+          <h1>Services</h1>
+          <UITable
+            ref="table"
+            data={this.props.services}
+            keyGetter={(service) => service.service.serviceId}
+            paginated={true}
+            rowChunkSize={15}
+          >
+            { StatusColumn }
+            { IdColumn }
+            { RequestColumn }
+            { GroupsColumn }
+            { UpstreamsColumn }
+            { ButtonsColumn }
```

Comment:
rather than rewriting this whole section inside the if statement, you could do something like this here instead:
```
{this.props.allowEdit && ButtonsColumn}
```